dlt/cli/magics.py

```diff
@@ -0,0 +1,288 @@
+
+
+
+@magics_class
+class DltMagics(Magics):
+
+    @property
+    def display(self) ->  t.Any:
+        if os.getenv("DATABRICKS_RUNTIME_VERSION"):
+            # Assume Databricks' 'display' is a callable with an unknown signature
+            databricks_display = self._shell.user_ns.get("display")
+            if callable(databricks_display):
+                return databricks_display
+            else:
+                raise RuntimeError("Expected a callable for Databricks' display, got something else.")
+        return display  # Assuming 'display' is a predefined callable
+
+    def success_message(self, messages: t.Dict[str, str]) -> HTML:
+        unstyled = messages.get("unstyled")
+        msg = str(
+            h(
+                "div",
+                h(
+                    "span",
+                    messages.get("green-bold"),
+                    {"style": {"color": "green", "font-weight": "bold"}},
+                ),
+                h("span", unstyled) if unstyled else "",
+            )
+        )
+        return HTML(msg)
+
+    def on_exception(self, ex: str, info: str) -> t.Any:
+        msg = str(
+            h(
+                "div",
+                h(
+                    "span",
+                    str(ex),
+                    {"style": {"color": "red", "font-weight": "bold"}},
+                ),
+                h(
+                    "span",
+                    " Please refer to %s for further assistance" % fmt.bold(info)
+                ),
+            )
+        )
+        out = display(HTML(msg))
+
+        if DEBUG_FLAG:
+            # Display the full traceback in a preformatted style
+            traceback_html = h("pre", traceback.format_exc(), {"style": {"color": "gray"}})
+            display(HTML(str(traceback_html)))
+            raise ex
+
+        return out
+
+
+    @magic_arguments()
+    @argument('--enable-telemetry', action='store_true', help="Enables telemetry before command is executed")
+    @argument('--disable-telemetry', action='store_true', help="Disables telemetry before command is executed")
+    @argument('--non-interactive', action='store_true', help="Non interactive mode. Default choices are automatically made for confirmations and prompts.")
+    @argument('--debug', action='store_true', help="Displays full stack traces on exceptions.")
+    @line_magic
+    @register_line_magic
+    def settings(self, line:str)->str:
+        """
+        A DLT line magic command to set global settings like telemetry, debug mode, etc.
+        """
+        args = parse_argstring(self.settings, line)
+        global DEBUG_FLAG
+        try:
+            from dlt.cli._dlt import telemetry_change_status_command_wrapper
+            if args.enable_telemetry:
+                telemetry_change_status_command_wrapper(True)
+            if args.disable_telemetry:
+                telemetry_change_status_command_wrapper(False)
+            if args.non_interactive:
```

Comment:
you can always switch non interactive with magics. and this is another way to get rid of all those special cases where notebook is non interactive

Reply:
please remove on interactive option - we force it anyway